### ISSUE TITLE
[BUG] - Setting up step > 1 and minValue has issues

### ISSUE THREAD

**Issue by omerman** [himelbrand/react-native-numeric-input#36]
First of all thank you for this library!

**Describe the bug**
When defining a step larger than 1 and also a minimum value,
once user presses minus, the onChange could be triggered with a value less than the minimum,
this could happen once though.

**To Reproduce**
1. step=5, minValue=0
2. set the input to 3
3. press minus
4. onchange is triggered with the value -2.

**and/or**

Share your relevant code segment. (usually, try and share at least your NumericInput component code)

```
        <NumericInput
          rounded
          step={5}
          minValue={0}
          value={value}
          onChange={(fieldValue) => {
            onValueChange(Number(fieldValue));
          }}
        />
```

**Expected behavior**
Expected behavior is that onChange will be triggered with the min value.

If any additional info is required, let me know. I don't believe it has anything to do with OS etc..

Thanks in advance!

**Comment by omerman** [comment]
For now I tried to apply my own fix.. checking inside the onChange itself.
And I also realise that I trigger onChange with 0, and update my store with 0, re render with 0, and I still get -2..Using this code:

```
<NumericInput
          rounded
          step={5}
          minValue={0}
          value={value}
          onChange={(fieldValue) => {
             if (value > minValue) {
                onValueChange(minValue);
              } else {
                onValueChange(fieldValue);
              }
          }}
        />
```

Which could only mean that even though I hold the state on my own, I get an uncontrolled component :X

**Comment by himelbrand** [comment]
@omerman thanks for reporting, I'll get to it as soon as possible

**Comment by himelbrand** [comment]
@omerman I fixed the issue.
it is now working as expected.
if there is still an issue please open again, closing.
the fix is available on version 1.8.1 on npm.

**Comment by eakeyson1** [comment]
@himelbrand I am currently facing this issue with version 1.8.3
